showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITHttpAnnotation.java

```diff
@@ -0,0 +1,104 @@
+    ComplianceSuite.Builder builder = ComplianceSuite.newBuilder();
+    JsonFormat.parser()
+        .merge(
+            new InputStreamReader(
+                ITHttpAnnotation.class
+                    .getClassLoader()
+                    .getResourceAsStream("compliance_suite.json")),
+            builder);
+    complianceSuite = builder.build();
+
+    ComplianceSettings httpjsonComplianceSettings =
+        ComplianceSettings.newHttpJsonBuilder()
+            .setCredentialsProvider(NoCredentialsProvider.create())
+            .setTransportChannelProvider(
+                ComplianceSettings.defaultHttpJsonTransportProviderBuilder()
+                    .setHttpTransport(
+                        new NetHttpTransport.Builder().doNotValidateCertificate().build())
+                    .setEndpoint("http://localhost:7469")
+                    .build())
+            .build();
+    complianceClient = ComplianceClient.create(httpjsonComplianceSettings);
+
+    // Mapping of Compliance Suite file RPC Names to ComplianceClient methods
+    complianceValidRpcMap =
+        ImmutableMap.of(
+            "Compliance.RepeatDataBody",
+            complianceClient::repeatDataBody,
+            "Compliance.RepeatDataBodyInfo",
+            complianceClient::repeatDataBodyInfo,
+            "Compliance.RepeatDataQuery",
+            complianceClient::repeatDataQuery,
+            "Compliance.RepeatDataSimplePath",
+            complianceClient::repeatDataSimplePath,
+            "Compliance.RepeatDataBodyPut",
+            complianceClient::repeatDataBodyPut,
+            "Compliance.RepeatDataBodyPatch",
+            complianceClient::repeatDataBodyPatch,
+            "Compliance.RepeatDataPathResource",
+            complianceClient::repeatDataPathResource);
+  }
+
+  @After
+  public void destroyClient() {
+    complianceClient.close();
+  }
+
+  /*
+  This test loops through all the testing groups (which test different aspects of the http annotation)
+  and calls the listed RPC Names (which are the different HttpVerbs) with the given request. The response's
+  info should be the same as the request's. This tests that the request is set correctly into the corresponding
+  URL/Body/Query.
+   */
+  @Test
+  public void testHttpAnnotations() {
+    for (ComplianceGroup compliancegroup : complianceSuite.getGroupList()) {
+      List<String> validRpcList =
+          compliancegroup.getRpcsList().stream()
+              .filter(complianceValidRpcMap::containsKey)
+              .collect(Collectors.toList());
+      for (String rpcName : validRpcList) {
+        Function<RepeatRequest, RepeatResponse> rpc = complianceValidRpcMap.get(rpcName);
+        for (RepeatRequest repeatRequest : compliancegroup.getRequestsList()) {
+          ComplianceData expectedData = repeatRequest.getInfo();
+          RepeatResponse response = rpc.apply(repeatRequest);
+          assertThat(response.getRequest().getInfo()).isEqualTo(expectedData);
```

Comment:
Could a helper method help reduce the number of nested for loops needed here? Reducing the amount of logic done in a test method can help with debugging in the future and prevent the test itself from developing bugs.

Reply:
Would it make more sense to just have each ComplianceGroup as its own test then? It could remove the outer most for loop if that's cleaner.

For each group, we test the product of rpcName and requestList.

Or did you mean something like:
```
  @Test
  public void testHttpAnnotations() {
    for (ComplianceGroup compliancegroup : complianceSuite.getGroupList()) {
      List<String> validRpcList =
          compliancegroup.getRpcsList().stream()
              .filter(complianceValidRpcMap::containsKey)
              .collect(Collectors.toList());
      helper(compliancegroup, validRpcList);
    }
  }

  private void helper(ComplianceGroup compliancegroup, List<String> validRpcList) {
    for (String rpcName : validRpcList) {
      Function<RepeatRequest, RepeatResponse> rpc = complianceValidRpcMap.get(rpcName);
      for (RepeatRequest repeatRequest : compliancegroup.getRequestsList()) {
        ComplianceData expectedData = repeatRequest.getInfo();
        RepeatResponse response = rpc.apply(repeatRequest);
        assertThat(response.getRequest().getInfo()).isEqualTo(expectedData);
      }
    }
  }
```

Reply:
I do like the idea of splitting the test based on the ComplianceGroup since it would reduce cognitive load needed in understanding the behavior the test is trying to verify. Also, in the event that a test fails it will be easier to determine where/what behavior is failing.  

Perhaps something similar to how [ComplianceClientTest](https://github.com/googleapis/gapic-generator-java/blob/820f4571af45b0d26ce3b883b27ebba5ac6f2270/showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/ComplianceClientTest.java#L41) is structured. 

Will also let @burkedavison and @blakeli0 weigh in.

Reply:
Sure, makes sense. I've split them by ComplianceGroup (unless there is also hesitancy about this). The groups are split by behavior (i.e. bindings, weird/extreme values, templating, etc.) so a failure for a test case in testing one group should alert us to the problematic behavior.